### DISCUSSION TITLE
Speed test consistency

### DISCUSSION THREAD

**Discussion by LarryT8959** [alexjustesen/speedtest-tracker#634] [Q&A]
I have speedtest tracker installed on a container on my Synology NAS. My question is what's causing my download speeds to fluctuate so much? It varies from 200mbps to 800. My upload speed remains above 900 all the time? I have 1GB service. Every time I check speed with a browser, it's above 900mbps? Is this network congestion? I have approx 60 devices connected in my smart home. What's the difference in the two test? I'm using the same test server. Trying to get to the bottom of this so that I can resolve any potential issues I may have. I've seem similar discussions but none say what could be my issue?
Thanks

**Comment by Erikk87** [comment]
Hi, I'm experiencing a similar issue: The results for speedtest-tracker are <100 MBps consistently, while using the speedtest.net web browser version, I get 900 MBps download speeds (I have a 1 GBit/s ISP). When testing the speed of the NAS network connection. Running a local speedtest b/w my laptop and the NAS, I'm betting close to 1 GBit/s, which is the speed limit of or LAN and thus the network card of the NAS should not be a bottleneck, nor should the ISP be one. With other dockerized services, I'm also getting good speeds of close to 1 GBit/s, so docker also (in general) does not seem to be the issue.

Any ideas? I'm happy to help investigating/debugging this ...

PS: I'm running the docker container on a Synology NAS DS216+II w/ 8 GB of RAM)

**Comment by LarryT8959** [comment]
Yeah, sounds similar. My Speedtest tracker results are all over the place? Using my MacBook Pro to the internet and to the NAS server I get over 900 consistently. I was thinking that my router was doing a poor job of managing traffic but if that were the case why when I run from browser I’m over 900 all the time? I’m thinking these eero routers are just struggling??Sent from my iPhoneOn Jun 16, 2023, at 9:29 AM, Erik ***@***.***> wrote:﻿
Hi, I'm experiencing a similar issue: The results for speedtest-tracker are <100 MBps consistently, while using the speedtest.net web browser version, I get 900 MBps download speeds (I have a 1 GBit/s ISP). When testing the speed of the NAS network connection. Running a local speedtest b/w my laptop and the NAS, I'm betting close to 1 GBit/s, which is the speed limit of or LAN and thus the network card of the NAS should not be a bottleneck, nor should the ISP be one. With other dockerized services, I'm also getting good speeds of close to 1 GBit/s, so docker also (in general) does not seem to be the issue.
Any ideas? I'm happy to help investigating/debugging this ...

**Reply by Erikk87** [reply]
I did a little bit of googling and it seems that the speedtest-cli is known for causing these kind of problems (= download speed results < 100 MBps). It seems that they are using a completely different engine for the CLI version compared to what is used on their website and in their desktop apps. Hence, I'm suspecting this to be the problem, as the CLI tool is also what is used at the core of speedtest-tracker. Also: I've used a different docker that has the plain CLI tool inside and again ended up with speeds < 100 MBps, which is even more evidence for that theory.

I guess, the only remedy for the problem would be Ookla to update their speedtest-cli, or switching to an entirely different speedtest engine in speedtest-tracker. Both not really easy-to-achieve options. :-(

**Reply by LarryT8959** [reply]
I guess for now, I won't rely on the results from my speedtest container, I'll pause for now. I was looking for something that would show me if I had a network issue. Getting results from 200 to 875 download, I was concerned there was a problem somewhere. Upload is always +900.

**Reply by Erikk87** [reply]
Ok, shame on me! I found the problem after testing a lot: the LAN cable b/w the NAS and the network switch/router caused the issue. Even though its a bit strange that the connection between local computers and the NAS (which is also running through the same switch/router) was fine, the download speed from the internet to the NAS was somehow slowed down. With a new cable, I'm getting decent speeds in and outside of docker containers.

**Comment by alexdelprete** [comment]
speedtest-cli works perfectly fine, I've been using it for years. And speedtest-tracker is working fine too, consistent results.

This is my least powerful system in the homelab, an RPi 4B+ with a 1Gb ethernet connection (never run tests via wifi), using speedtest-cli now on my 1Gb symmetrical fiber:

```
> speedtest -s 4302

   Speedtest by Ookla

      Server: Aruba S.p.A. - Ponte San Pietro (id: 40519)
         ISP: Tecno General S.r.l
Idle Latency:    12.05 ms   (jitter: 1.91ms, low: 11.51ms, high: 14.76ms)
    Download:   929.30 Mbps (data used: 1.0 GB)
                127.64 ms   (jitter: 62.50ms, low: 11.68ms, high: 592.48ms)
      Upload:   756.77 Mbps (data used: 1.3 GB)
                 13.81 ms   (jitter: 1.63ms, low: 11.65ms, high: 20.12ms)
 Packet Loss:     0.0%

```

I would advice to install speedtest-cli on a linux pc or VM, and run it from there. If you have an RPi, it's perfect for the test. Run speedtest using the -s option, specifying the same server ID you use on their website, so you can compare apples to apples. If you don't specify the server, it will be randomly chosen, so make sure you're using the same server when you test.

**Comment by LarryT8959** [comment]
I don’t have a RPI, just a Mac and a NAS?

**Comment by LarryT8959** [comment]
I restarted my NAS for the first time in over a couple months and the speeds look to be more consistent currently. They are 100 slower on the download, and right on for upload. I will attempt to find a docker package for speediest cli.
Thanks for everyone’s input.

**Reply by alexdelprete** [reply]
Install speedtest-cli on your Mac: `https://www.speedtest.net/apps/cli`

![image](https://github.com/alexjustesen/speedtest-tracker/assets/7027842/7e48cf5d-5516-4331-9cde-672e867bfcb1)

Then find the server id you test with speedtest-tracker, and use it via CLI (option -s), so you can compare results properly. Use an ethernet connection, not wifi.

What NAS model do you have?

**Comment by LarryT8959** [comment]
I don’t have home-brew installed so I just downloaded the file and unzipped to run. These are the results I got, and it was on same server.


￼

**Reply by alexdelprete** [reply]
I don't see any screenshot...just paste the text of the output in the shell...

**Comment by LarryT8959** [comment]
￼

**Comment by LarryT8959** [comment]
I’ll try some new cables, they should deliver tomorrow. I’ll let you know what happens.
Thanks

**Comment by LarryT8959** [comment]
￼￼

**Reply by alexdelprete** [reply]
No screenshots...I see "OBJ". Just paste the text...no image.

![image](https://github.com/alexjustesen/speedtest-tracker/assets/7027842/5d245b8a-ca35-4509-9c98-820842311ec0)

**Comment by LarryT8959** [comment]
Speedtest by Ookla
Server: Magna5 - Manassas, VA (id: 55084)
ISP: Verizon Fios
Idle Latency:
6.94 Ms (jitter: 0.16ms, low: 6.88m, high: 7.27ms)
Download:
756.42 Mbps (data used: 887.0 MB)
Upload:
7.58 ms (jitter: 0.33ms, low: 7.30ms, high: 15.15ms)
864.21 Mbps (data used: 953.1 MB)
8.16 ms
0.0%
(jitter: 0.61ms, low: 7.37ms, high: 17.46ms)
Packet Loss:
8e643
Result URL: https: / /www.speedtest.net/result /c/c3592c5b-7aa9-4451-a1a1-f6fe9a7
Saving session..
..copying shared history..
.. saving history ...truncating history files..
.. completed.
[Process completed]926.12Ping ms4 8ConnectionsMultiMagna5Manassas, VAChange ServerVerizon® UPLOAD Mbps923.50@ 8©11RATE YOUR PROVIDERVerizonSent from my iPhoneOn Jun 18, 2023, at 9:30 AM, Alessandro Del Prete ***@***.***> wrote:﻿
No screenshots...I see "OBJ". Just paste the text...no image.

**Reply by alexdelprete** [reply]
So on Mac with ethernet and with speedtest-cli you have 756/864. Not bad. As you can see, speedtest-cli works perfectly fine.

So your issue is on the NAS: either docker or the network of the NAS. Concentrate on that.

**Comment by LarryT8959** [comment]
I’ve increased the MTU value on both interfaces. No completely sure what that does but it has really helped. Download speed is up over 900 on four consecutive test. Never seen that before.
No need for new cables.

**Reply by alexdelprete** [reply]
On the NAS or on the Mac? In any case: NEVER mess with the MTU. It seems to help, but it also leads to weird issues because ALL the networking devices in the chain have to support large MTU values.

Don't mess with that, and use defaults, unless you are a networking expert.

And if you have modified MTU on the NAS, it could be the cause of the issues you were having with speedtests.

**Comment by LarryT8959** [comment]
On the NAS. I’m certainly no expert but it has solved the problem?

**Comment by alexdelprete** [comment]
No, it didn't. The default MTU value should work fine. Higher MTU values are needed only in specific cases, and all devices on the network chain (switches and other clients of the NAS) have to have the same MTU value.

Try to set MTU to default again on the NAS and retry.

**Comment by LarryT8959** [comment]
Back down to 740 download after setting to default. I’ve put a ticket into Synology.

**Comment by alexdelprete** [comment]
> Back down to 740 download after setting to default. I’ve put a ticket into Synology.

Does it fluctuate as per your original post? If it's constant 740, it's a good sign. Now you just have to understand with Synology why it doesn't squeeze more speed...but that has nothing to do with this project.

NAS devices have a custom OS with custom tuned settings. That's why I don't recommend using them for these kind of applications.

**Comment by LarryT8959** [comment]
I understand, I just want everything performing at a high level. If Synology tells me that 740 is about as good as it gets, then I’ll accept it. To me, it should download at 900 like everything else on my network. They may also tell me to leave the MTU alone? Also it could be something else I can do. I’ll wait to see what they have to say.
What I know now its definitely a NAS issue.
Thanks

**Comment by LarryT8959** [comment]
Changed MTU back to 9000 and speed is back up to 940?

**Reply by alexdelprete** [reply]
What speed do you have with the Mac via ethernet? 940?

I'm just warning you that changing MTU might lead to subtle issues in the future, that's why it's not recommended changing it, unless there's a specific reason. And if you decide to change it, the switch to which the NAS is connected (or the router) and the NAS clients (other PCs, etc.) have to be properly configured too.

**Comment by LarryT8959** [comment]
940,I understand what you’re saying. I’d like to see 900 all the time on everything. Maybe at the end of the day that’s not realistic. Let’s see what Synology have to say.Thanks Sent from my iPhoneOn Jun 18, 2023, at 2:34 PM, Alessandro Del Prete ***@***.***> wrote:﻿
What speed do you have with the Mac via ethernet? 940?
I'm just warning you that changing MTU might lead to subtle issues in the future, that's why it's not recommended changing it, unless there's a specific reason. And if you decide to change it, the switch to which the NAS is connected (or the router) and the NAS clients (other PCs, etc.) have to be properly configured too.

**Comment by alexdelprete** [comment]
So the fact that on the Mac, with the standard MTU, you reach the full speed confirms what I'm saying: jumbo frames (MTU 9000) are not needed.

Try to find speedtest-cli for the synology, so you can test from shell vs docker. Maybe the problem is the docker on Synology.

**Comment by LarryT8959** [comment]
I’ve been running MTU at 9000 for over a week and haven’t noticed any issues so far and my speeds are better but not perfect like other clients. Here are a couple notes from Synology.

That's great to see that its at least working for you. It's hard to gaurentee consistancy when it comes to network speeds on a diskstation. They always fluctuate especitally when you have several services and packages that your DS220+ may be using and the type of network you are on. But I would like to note that those speeds you have currently are good.


There is no downside to setting your Diskstation's MTU value to 9000, as long as all devices in the network support jumbo frames. However, it is important to note that some built-in 1GbE interfaces may not support an MTU of 9000.

**Comment by alexdelprete** [comment]
"it's hard to guarantee consistency"...on a NAS that costs 4 times a PC with disks? :)

If a device needs jumbro frames to do 1Gb/s on its ethernet port, it means its firmware (OS and config) is not really optimized.

Anyway, you know what the drawbacks of jumbo frames are...keep it like you prefer while it's working.

**Comment by LarryT8959** [comment]
Good point,
The NAS defiantly has limitations but it seemed to be a pretty good investment for now. For a little over $200 for the server, and another $350 for the  14TB of hard drive capacity and  a systems that stays on all the time running containers and software, I don’t think its too bad. Its does a pretty good job for my photography workflow also.
As I grow and expand maybe I’ll setup one of my older Macs as a server with some external drives?

Thanks

**Comment by alexdelprete** [comment]
$200 is a great deal. But don't expect miracles. :)

**Comment by LarryT8959** [comment]
Hello,
I’m starting to some signs of a communication issue with me NVR for security cameras. Every now and then I hear a beep from time to time. Per Hikvision, the cause could be hard drive, or network. I’ve heard this before but not this often. I’m switching mu’s back to normal. I’ll let you know if the beeping will stop.

On another note, I need some advice. I have an older  MAC that I want to put to use. I would like to run some of my home automation stuff there also? What do you think is the best way to handle this?
Install hoobs
Install Homebridge
Install docker desktop
Add a virtual machine?

I’ve added docker desktop but I thought it would be like the docker on my NAS. It’s easy to download packages. This is coding from what I can tell, learning curve???

Thanks

**Reply by alexjustesen** [reply]
You're absolutely going to have a learning curve when it comes to running docker and especially taking the UI out of the equation. Can't speak for MACs but I run my home lab services on a few different devices include raspberry pis, intel nucs, synology nas and an unraid nas.

I'd probably recommend those over a MAC
